assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java

```diff
@@ -412,4 +428,68 @@ private void configureJwtProxySecureProvisioner(String infrastructure) {
           .to(new TypeLiteral<JwtProxySecureServerExposerFactory<OpenShiftEnvironment>>() {});
     }
   }
+
+  private void configureFakeJwtProxySecureProvisioner(String infrastructure) {
+    if (KubernetesInfrastructure.NAME.equals(infrastructure)) {
+      MapBinder.newMapBinder(
+              binder(),
+              new TypeLiteral<String>() {},
+              new TypeLiteral<SecureServerExposerFactory<KubernetesEnvironment>>() {})
+          .addBinding("jwtproxy")
+          .to(
+              new TypeLiteral<
+                  PassThroughProxySecureServerExposerFactory<KubernetesEnvironment>>() {});
+    } else {
+      MapBinder.newMapBinder(
+              binder(),
+              new TypeLiteral<String>() {},
+              new TypeLiteral<SecureServerExposerFactory<OpenShiftEnvironment>>() {})
+          .addBinding("jwtproxy")
+          .to(
+              new TypeLiteral<
+                  PassThroughProxySecureServerExposerFactory<OpenShiftEnvironment>>() {});
+    }
+  }
+
+  private void installDefaultSecureServerExposer(String infrastructure) {
+    if (KubernetesInfrastructure.NAME.equals(infrastructure)) {
+      MapBinder<String, SecureServerExposerFactory<KubernetesEnvironment>>
+          secureServerExposerFactories =
+              MapBinder.newMapBinder(binder(), new TypeLiteral<>() {}, new TypeLiteral<>() {});
+
+      install(
+          new FactoryModuleBuilder()
+              .implement(
+                  new TypeLiteral<SecureServerExposer<KubernetesEnvironment>>() {},
+                  new TypeLiteral<PassThroughProxySecureServerExposer<KubernetesEnvironment>>() {})
+              .build(
+                  new TypeLiteral<
+                      PassThroughProxySecureServerExposerFactory<KubernetesEnvironment>>() {}));
+
+      secureServerExposerFactories
+          .addBinding("default")
+          .to(
+              new TypeLiteral<
+                  PassThroughProxySecureServerExposerFactory<KubernetesEnvironment>>() {});
+    } else {
+      MapBinder<String, SecureServerExposerFactory<OpenShiftEnvironment>>
```

Comment:
Looks like you can do 
`MapBinder<String, SecureServerExposerFactory<? extends KubernetesEnvironment>>`, can't you?
to move map out of `if` cond.

Reply:
would be nice if possible

Reply:
Unfortunately this wouldn't work because of type literals - you need the full generic type params to be able express the bindings, so there is going to be an if sooner or later somewhere. I've played with this and couldn't come up with anything that would improve the code in any reasonable way.